src/worker.js

```diff
@@ -203,7 +203,9 @@ this.onmessage = function(e) {
         } else {
           Atomics.store(HEAPU32, (threadInfoStruct + 4 /*C_STRUCTS.pthread.threadExitCode*/ ) >> 2, (e instanceof {{{ makeAsmGlobalAccessInPthread('ExitStatus') }}}) ? e.status : -2 /*A custom entry specific to Emscripten denoting that the thread crashed.*/);
```

Comment:
Good point about the wrong stack trace being shown.

How about this: At the top of this `else` body, we know that this isn't an infinite loop etc., and then we want to do some changes like these Atomics and the futex stuff. How about if before the atomics, we check if we managed to set up the thread properly, that is, move your if to up here, and put it behind ASSERTIONS. So something like
```
#if ASSERTIONS
    if (typeof({{{ makeAsmGlobalAccessInPthread('_emscripten_futex_wake') }}}) !== "function") {
      err("could not set up thread properly");
      throw e;
   }
#endif
```

Reply:
Done.